src/engagement_db_to_analysis/engagement_db_to_analysis.py

```diff
@@ -112,26 +150,74 @@ def generate_analysis_files(user, google_cloud_credentials_file_path, pipeline_c
 
     run_automated_analysis(messages_by_column, participants_by_column, pipeline_config.analysis, f"{output_dir}/automated-analysis")
 
+
+def generate_analysis_files(user, google_cloud_credentials_file_path, pipeline_config, uuid_table, engagement_db, rapid_pro,
+                            membership_group_dir_path, output_dir, cache_path=None, dry_run=False):
+    """
+    :type pipeline_config: src.pipeline_configuration_spec.PipelineConfiguration
+    """
+
+    analysis_dataset_configurations = pipeline_config.analysis.dataset_configurations
+    # TODO: Tidy up which functions get passed analysis_configs and which get passed dataset_configurations
+
+    if cache_path is None:
+        cache = None
+        log.warning(f"No `cache_path` provided. This tool will perform a full download of project messages from engagement database")
+    else:
+        log.info(f"Initialising EngagementAnalysisCache at '{cache_path}/engagement_db_to_analysis'")
+        cache = AnalysisCache(f"{cache_path}/engagement_db_to_analysis")
+
+    engagement_db_datasets = []
+    for config in analysis_dataset_configurations:
+        engagement_db_datasets.extend(config.engagement_db_datasets)
+    messages_map = get_messages_in_datasets(engagement_db, engagement_db_datasets, cache, dry_run)
+
+    messages = [message for messages_list in messages_map.values() for message in messages_list]
```

Comment:
Can you extract this to a function or convert to a nested for loop? I think list comps are quite hard to read when they have multiple `for`s